src/data/finset/noncomm_prod.lean

```diff
@@ -235,6 +257,30 @@ by simp [noncomm_prod, insert_val_of_not_mem ha, multiset.noncomm_prod_cons']
     (λ x hx y hy, by rw [mem_singleton.mp hx, mem_singleton.mp hy]) = f a :=
 by simp [noncomm_prod, multiset.singleton_eq_cons]
 
+@[to_additive]
+lemma _root_.submonoid.finset_noncomm_prod_mem (s : finset α) (f : α → β)
+  (comm : ∀ (x : α), x ∈ s → ∀ (y : α), y ∈ s → commute (f x) (f y))
+  (m : submonoid β) (h : ∀ (x : α), x ∈ s → f x ∈ m) : s.noncomm_prod f comm ∈ m :=
+begin
+  apply submonoid.multiset_noncomm_prod_mem,
+  intro y,
+  rw multiset.mem_map,
+  rintros ⟨x, ⟨hx, rfl⟩⟩,
+  exact h x hx,
+end
+
+@[to_additive]
+lemma _root_.subgroup.finset_noncomm_prod_mem {β : Type*} [group β] (s : finset α) (f : α → β)
+  (comm : ∀ (x : α), x ∈ s → ∀ (y : α), y ∈ s → commute (f x) (f y))
+  (m : subgroup β) (h : ∀ (x : α), x ∈ s → f x ∈ m) : s.noncomm_prod f comm ∈ m :=
+begin
+  apply subgroup.multiset_noncomm_prod_mem,
+  intro y,
+  rw multiset.mem_map,
+  rintros ⟨x, ⟨hx, rfl⟩⟩,
+  exact h x hx,
+end
```

Comment:
Can you get here faster with `m.to_submonoid.finset_noncomm_prod_mem`?

Reply:
Ah, probably! I'll try. I think I disregarded it because when I came across this problem first it involved the `bsupr` of subgroups, and the `.range`,  and there I didn't find the lemmas that let me reduce this to the world of submonoids. But for this simple lemma you are right.

Reply:
done. Also reordered arguments to allow dot notation.